packages/shared/src/components/onboarding/OnboardingFeedHeader.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useCallback, useEffect } from 'react';
+import classNames from 'classnames';
+import { useRouter } from 'next/router';
+import { FilterOnboardingV4 } from './FilterOnboardingV4';
+import {
+  Button,
+  ButtonColor,
+  ButtonIconPosition,
+  ButtonSize,
+  ButtonVariant,
+} from '../buttons/Button';
+import { REQUIRED_TAGS_THRESHOLD } from './common';
+import { ArrowIcon } from '../icons';
+import { LayoutHeader } from '../layout/common';
+import { usePrompt } from '../../hooks/usePrompt';
+import { useMyFeed } from '../../hooks/useMyFeed';
+import { useAlertsContext } from '../../contexts/AlertContext';
+import { useAnalyticsContext } from '../../contexts/AnalyticsContext';
+import { AnalyticsEvent } from '../../lib/analytics';
+
+const promptConfig = {
+  title: 'Discard tag selection?',
+  description: 'Your personalized feed is only a few tags away',
+  cancelButton: {
+    title: 'Stay',
+    variant: ButtonVariant.Primary,
+    color: ButtonColor.Cabbage,
+  },
+  okButton: {
+    title: 'Leave',
+    variant: ButtonVariant.Secondary,
+  },
+  className: {
+    buttons: 'flex-row-reverse',
+  },
+};
+
+type OnboardingFeedHeaderProps = {
+  isPreviewFeedVisible: boolean;
+  setPreviewFeedVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  isFeedPreviewEnabled: boolean;
+  tagsCount: number;
+};
+
+export const OnboardingFeedHeader = ({
+  isPreviewFeedVisible,
+  setPreviewFeedVisible,
+  isFeedPreviewEnabled,
+  tagsCount,
+}: OnboardingFeedHeaderProps): JSX.Element => {
+  const { updateAlerts } = useAlertsContext();
+  const { trackEvent } = useAnalyticsContext();
+  const { showPrompt } = usePrompt();
+  const router = useRouter();
+  const { registerLocalFilters } = useMyFeed();
+
+  const completeOnboarding = useCallback(() => {
+    registerLocalFilters();
+    updateAlerts({ filter: false, myFeed: 'created' });
+
+    trackEvent({
+      event_name: AnalyticsEvent.CreateFeed,
+    });
+
+    return router.replace({
+      pathname: router.route,
+      query: {
+        welcome: 'true',
+      },
+    });
+  }, [registerLocalFilters, router, trackEvent, updateAlerts]);
+
+  useEffect(() => {
+    let stopNav = true;
+
+    const routeHandler = (newRoute: string) => {
+      if (!stopNav || newRoute === '/my-feed') {
+        return;
+      }
+      router.events.emit('routeChangeError');
+
+      showPrompt(promptConfig).then((leave) => {
+        if (leave) {
+          stopNav = false;
+          trackEvent({
+            event_name: AnalyticsEvent.OnboardingSkip,
+          });
```

Comment:
Not sure if this is right here?
Was it checked with Dave/Ido?

I think we might want one, but a new one?

Reply:
I'll check with Dave

Reply:
Will check with Dave

Reply:
We'll re-use